### PR TITLE
Local setup under macOs

### PR DESCRIPTION
**Pull request by lockejan**
another attempt to resolve https://github.com/NixOS/nixos-homepage/issues/515

After https://github.com/NixOS/nixos-homepage/pull/584 has been closed, I just tested it again and it still works.
The README on master evolved a bit since then, which is why I cleaned up the last proposal of @bxrt a bit.

It's not a completely outlined scenario how to hack on the project under macOS but it should suffice to get started @garbas .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,6 +63,35 @@ substituters = ... https://nixos-homepage.cachix.org
...
+
+    `$ docker pull nixos/nix`
+
+2. Start and enter a container:
+
+    `$ docker run -p 8000:8000 -it --name nixos-homepage nixos/nix`
+
+3. (Optional) Having entr, fd and vim at place is helpful for automatic rebuilds of the site and enabling binary cache as described in [non-NixOs](#on-non-nixos).
+
+    `nix-shell -p entr fd vim`
```
Comment (garbas) [inline README.md:81]:
```suggestion
    `nix-shell`
```

The `serve` script takes care of reloading, rebuilding and serving this project. No need to use any other tool.

File: README.md [hunk 2]
```diff
@@ -63,6 +63,35 @@ substituters = ... https://nixos-homepage.cachix.org
...
+
+4. To prepare and serve the project:
+
+    ```
+    $ git clone https://github.com/NixOS/nixos-homepage.git
+    $ cd nixos-homepage
+    $ nix-shell
+    [nix-shell]$ make
```
Comment (garbas) [inline README.md:89]:
```suggestion
    [nix-shell]$ serve
```

Is `serve` not working on MacOS?

File: README.md [hunk 3]
```diff
@@ -63,6 +63,35 @@ substituters = ... https://nixos-homepage.cachix.org
...
+    [nix-shell]$ python -m http.server --bind 0.0.0.0
```
Comment (garbas) [inline README.md:90]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -63,6 +63,35 @@ substituters = ... https://nixos-homepage.cachix.org
...
+    ```
+
+    Note: to continuously rebuild the site on changes use `[nix-shell]$ fd | entr make` instead.
```
Comment (garbas) [inline README.md:93]:
```suggestion
```

**Review by garbas — CHANGES_REQUESTED** [review]
Thank you for looking into this. Not sure if you know but there is now a `serve` script which you can run that handles reload, rebuild and serving of the website.

File: README.md (hunk 2 above)
Comment (lockejan) [reply README.md:89]: Serve is working inside the container. But it only binds to localhost, so request from the mac host aren't picked up😬

**Comment by lockejan** [comment]
> Thank you for looking into this. Not sure if you know but there is now a `serve` script which you can run that handles reload, rebuild and serving of the website. Can you check if it works on MacOS?

As long as it helps, you're welcome.
I haven't noticed `serve` till you mentioned it to be honest.
As responded to your suggestion already, `serve` works inside the container, but not natively on my darwin machine.

I'm using direnv, the shellHook fails on the current master branch but not on my feat-branch. However it doesn't make a difference for `nix-shell` and `nix shell`. Both are failing.

`nix-shell` results in 
>error: nix-shell requires a single derivation

and 

`nix shell` results (marked it as trusted on first invocation already) in 
>warning: Using saved setting for 'substituters = https://nixos-homepage.cachix.org https://nixos-nix-install-tests.cachix.org' from ~/.local/share/nix/trusted-settings.json.
warning: Using saved setting for 'trusted-public-keys = nixos-homepage.cachix.org-1:NHKBt7NjLcWfgkX4OR72q7LVldKJe/JOsfIWFDAn/tE= nixos-nix-install-tests.cachix.org-1:Le57vOUJjOcdzLlbwmZVBuLGoDC+Xg2rQDtmIzALgFU=' from ~/.local/share/nix/trusted-settings.json.
warning: ignoring untrusted substituter 'https://nixos-homepage.cachix.org'
warning: ignoring untrusted substituter 'https://nixos-nix-install-tests.cachix.org'
error: flake 'git+file:///Users/lockejan/git/nixos-homepage' does not provide attribute 'packages.x86_64-darwin.default' or 'defaultPackage.x86_64-darwin'

**Comment by lockejan** [comment]
I don't have time to further look into this right now. Feel free to close this @garbas

**Pull request closed without merging**